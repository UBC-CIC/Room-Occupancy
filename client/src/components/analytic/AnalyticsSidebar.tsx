import * as React from "react";
import { Button, Icon, List, SemanticICONS } from "semantic-ui-react";
import { Link } from "react-router-dom";
import { useAuthenticator } from "@aws-amplify/ui-react";

import "../../pages/analytics/Analytics.css";

export interface IAnalyticsSidebarProps {}

interface IAnalyticsItems {
  category: analyticType;
  name: string;
  icon: SemanticICONS;
  link: string;
}

export enum analyticType {
  analytics = "analytics",
  locations = "location",
}

// add values for side bar here
const analyticsItems: IAnalyticsItems[] = [
  {
    category: analyticType.analytics,
    name: "Analytics",
    icon: "chart area",
    link: "/analytics",
  },
  {
    category: analyticType.analytics,
    name: "Camera",
    icon: "camera",
    link: "/analytics",
  },
  {
    category: analyticType.locations,
    name: "Zone 1",
    icon: "location arrow",
    link: "/analytics",
  },
  {
    category: analyticType.locations,
    name: "Zone 2",
    icon: "location arrow",
    link: "/analytics",
  },
];

export function AnalyticsSidebar(props: IAnalyticsSidebarProps) {
  const { authStatus } = useAuthenticator((context) => [context.authStatus]);
  return (
    <List>
      {authStatus === "authenticated" && (
        <Link to="/admin_dashboard">
          <Button color="blue" fluid icon labelPosition="left">
            <Icon name="dashboard" />
            Dashboard
          </Button>
        </Link>
      )}
      <br />
      <List.Item>
        <List.Header>ANALYTICS</List.Header>
        <List.Content>
          <List>
            {analyticsItems
              .filter((item) => item.category === analyticType.analytics)
              .map((value, key) => {
                return (
                  <Link to={value.link}>
                    <List.Item key={key}>
                      <List.Icon color="grey" name={value.icon} />
                      <List.Content as="a">{value.name}</List.Content>
                    </List.Item>
                  </Link>
                );
              })}
          </List>
        </List.Content>
      </List.Item>
      <List.Item>
        <List.Header>LOCATIONS</List.Header>
        <List.Content>
          <List>
            {analyticsItems
              .filter((item) => item.category === analyticType.locations)
              .map((value, key) => {
                return (
                  <Link to={value.link}>
                    <List.Item key={key}>
                      <List.Icon color="grey" name={value.icon} />
                      <List.Content as="a">{value.name}</List.Content>
                    </List.Item>
                  </Link>
                );
              })}
          </List>
        </List.Content>
      </List.Item>
    </List>
  );
}
