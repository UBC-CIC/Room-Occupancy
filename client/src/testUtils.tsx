import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import {
  MemoryRouter,
  MemoryRouterProps,
  BrowserRouter,
} from "react-router-dom";

// Implement with creation of providers
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Usage example:
// test("matches snapshot", () => {
//   // Replace 'Navbar' with your desired component and provide the props
//   testSnapshot(Navbar, { /* props for Navbar */ });
// });
const testComponentSnapshot = <P extends {}>(
  Component: React.ComponentType<P>,
  props: P,
  routerProps: MemoryRouterProps = {}
) => {
  const { asFragment } = render(
    <MemoryRouter {...routerProps}>
      <Component {...props} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
};

export * from "@testing-library/react";
export { customRender, testComponentSnapshot };
