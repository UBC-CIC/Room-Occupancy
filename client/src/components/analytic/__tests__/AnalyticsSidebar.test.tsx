import React from "react";
import "@testing-library/jest-dom";
import { customRender } from "../../../testUtils";
import { AnalyticsSidebar } from "../AnalyticsSidebar";
import { screen } from "../../../testUtils";

describe("<AnalyticsSidebar />", () => {
  test("should matchSnapshot", () => {
    const { asFragment } = customRender(
      <AnalyticsSidebar>Colin</AnalyticsSidebar>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render the correct headers", () => {
    const headers = ["ANALYTICS", "LOCATIONS"];
    const { asFragment } = customRender(
      <AnalyticsSidebar>Colin</AnalyticsSidebar>
    );

    headers.map((header) => {
      expect(screen.getByText(header)).toBeInTheDocument();
    });
  });
});
