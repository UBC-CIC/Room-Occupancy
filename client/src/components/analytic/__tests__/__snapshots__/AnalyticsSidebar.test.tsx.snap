// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyticsSidebar /> should matchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="ui list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <div
        class="header"
      >
        ANALYTICS
      </div>
      <div
        class="content"
      >
        <div
          class="ui list"
          role="list"
        >
          <div
            class="item"
            role="listitem"
          >
            <i
              aria-hidden="true"
              class="grey chart area icon"
            />
            <a
              class="content"
            >
              Analytics
            </a>
          </div>
          <div
            class="item"
            role="listitem"
          >
            <i
              aria-hidden="true"
              class="grey camera icon"
            />
            <a
              class="content"
            >
              Camera
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <div
        class="header"
      >
        LOCATIONS
      </div>
      <div
        class="content"
      >
        <div
          class="ui list"
          role="list"
        >
          <div
            class="item"
            role="listitem"
          >
            <i
              aria-hidden="true"
              class="grey location arrow icon"
            />
            <a
              class="content"
            >
              Zone 1
            </a>
          </div>
          <div
            class="item"
            role="listitem"
          >
            <i
              aria-hidden="true"
              class="grey location arrow icon"
            />
            <a
              class="content"
            >
              Zone 2
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
